UTRannotator.pm

```diff
@@ -0,0 +1,1306 @@
+                        "uSTOP_lost_Evidence" => $uSTOP_lost_evidence,
+                    );
+
+                    $output_flag = $output_flag? $output_flag."&"."uSTOP_lost":"uSTOP_lost";
+                    $output_effects = $output_effects? $output_effects."&".$self->transform_hash_to_string(\%uORF_effect):$self->transform_hash_to_string(\%uORF_effect);
+
+                }
+
+
+
+
+        }
+    }
+
+    $result{'uSTOP_lost_flag'} = $output_flag;
+    $result{'uSTOP_lost_effect'} = $output_effects;
+
+    return \%result;
+}
+
+sub uAUG_lost {
+    # Description: annotate if a five_prime_UTR_varint removes a start codon of an existing uORF
+
+    my ($self, $variant_info, $UTR_info, %kozak_strength) = @_;
+
+    my $chr = $variant_info->{chr};
+    my $pos = $variant_info->{pos};
+    my $ref = $variant_info->{ref};
+    my $alt = $variant_info->{alt};
+
+    my @sorted_starts = @{$UTR_info->{start}};
+    my @sequence = split //, $UTR_info->{seq};
+    my $utr_length = @sequence;
+    my $strand = $UTR_info->{strand};
+
+    my %existing_ref_uORF = %{$self->existing_uORF(\@sequence)};
+
+    #return annotators
+    my $uAUG_lost_type = "";   # the uORF type with the reference allele - uORF, inframe_oORF, outOfFrame_oORF
+    my $uAUG_lost_DistanceToCDS = ""; # the distance of the lost uAUG to CDS
+    my $uAUG_lost_DistanceToSTOP = ""; # the distance of the lost uAUG to stop codon (could be in CDS)
+    my $uAUG_lost_KozakContext = ""; # the Kozak context sequence of the lost uAUG
+    my $uAUG_lost_KozakStrength = ""; # the strength of KozakContext of the lost uAUG
+    my $uAUG_lost_evidence = ""; # whether this uAUG has any evidence of being translated
+
+    my $current_kozak = "";
+    my $current_kozak_strength = "";
+
+    #indicate whether the variant ever disrupts a stop codon
+    my $output_flag = "";
+    #indicate whether the variant ever disrupts the uORF that being analyzed
+    my $flag_uORF=0;
+
+    #the relative position of input variant in the UTR sequence
+
+
+    my %result = (); #result is a hash table with two elements: $flag and $output_effects
+    my $output_effects="";
+
+    my $ref_coding = $self->get_ref_coding($ref);
+    my $alt_coding = $self->get_alt_coding($alt,$strand);
+
+    my ($mut_pos, $end_pos) = $self->get_allele_exon_pos($strand, $pos, $ref_coding, $UTR_info);
+    return {} unless(defined($mut_pos)&defined($end_pos));
+
+
+	my $mut_utr_seq = $self->mut_utr_sequence(\@sequence,$mut_pos,$ref_coding,$alt_coding,$strand);
+  	my @mut_utr_seq = split //,$mut_utr_seq;
+
+    my @start = @{$self->get_ATG_pos(\@sequence)};
+
+
+    for (my $i=0;$i<@start;$i++){
+
+        $flag_uORF=0;
+        my $start_pos = $start[$i];
+
+
+        next if ($mut_pos-$start_pos>2);
+
+        if (length($ref_coding) ne 0){
+            next if ($mut_pos+length($ref_coding)-1<$start_pos);
+        }
+        next if ($mut_pos<$start_pos);   #for insertion
+
+        #for deletion, it definitely disrupting the uORF.
```

Comment:
It should be?
#for deletion, it is definitely disrupting the uORF.

Reply:
Not sure.